### PR TITLE
[ Bug ] Fix calculating training loss

### PR DESCRIPTION
**Pull request by jijoongmoon**
In this PR, calculation of training loss is fixed.
Previously, network loss is calculated using last mini_batch size data set.
This should be summarized through out whole data set and averaged.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
Wow.. you finally got that bug. Great!

**Comment by myungjoo** [comment]
It'd be better if you'd describe what was the problem with the previous code, though.

File: nntrainer/src/neuralnet.cpp [hunk 1]
```diff
@@ -701,17 +693,16 @@ int NeuralNetwork::train_run() {
         }
         std::cout << "#" << i + 1 << "/" << epoch;
         data_buffer->displayProgress(count++, nntrainer::BUF_TRAIN, getLoss());
+        training_loss += getLoss();
```
Comment (kparichay) [inline nntrainer/src/neuralnet.cpp:696]: This is good catch.

File: nntrainer/src/neuralnet.cpp [hunk 2]
```diff
@@ -575,14 +575,7 @@ void NeuralNetwork::backwarding(sharedConstTensor input,
...
 
-float NeuralNetwork::getLoss() {
-  loss = 0.0f;
-  for (unsigned int i = 0; i < layers.size(); i++) {
-    loss += layers[i]->getLoss();
-  }
-
-  return loss;
-}
+float NeuralNetwork::getLoss() { return layers.back()->getLoss(); }
```
Comment (kparichay) [inline nntrainer/src/neuralnet.cpp:578]: This will ignore the loss by the weight decay loss. Is this intentional?

Comment (jijoongmoon) [reply nntrainer/src/neuralnet.cpp]: fixed.
